book_maker/loader/epub_loader.py

```diff
@@ -89,12 +113,44 @@ def make_bilingual_book(self):
         index = 0
         p_to_save_len = len(self.p_to_save)
         try:
+            # Add the things that don't need to be translated first, so that you can see the img after the interruption
             for item in self.origin_book.get_items():
-                if item.get_type() == ITEM_DOCUMENT:
-                    soup = bs(item.content, "html.parser")
-                    p_list = soup.findAll(trans_taglist)
-                    if self.allow_navigable_strings:
-                        p_list.extend(soup.findAll(text=True))
+                if item.get_type() != ITEM_DOCUMENT:
+                    new_book.add_item(item)
+
+            for item in self.origin_book.get_items_of_type(ITEM_DOCUMENT):
+                soup = bs(item.content, "html.parser")
+                p_list = soup.findAll(trans_taglist)
+                if self.allow_navigable_strings:
+                    p_list.extend(soup.findAll(text=True))
+
+                sendNum = self.accumulated_num
+                if sendNum > 1:
+                    count = 0
+                    waitPList = []
+                    for i in range(0, len(p_list)):
+                        p = p_list[i]
+                        if not p.text or self._is_special_text(p.text):
+                            continue
+                        length = len(p.text)
+                        if length > sendNum:
+                            waitPList = deal_new(p, waitPList)
+                            continue
+                        if i == len(p_list) - 1:
+                            if count + length < sendNum:
+                                waitPList.append(p)
+                                waitPList = deal_old(waitPList)
+                            else:
+                                waitPList = deal_new(p, waitPList)
+                            break
+                        if count + length < sendNum:
+                            count += length
+                            waitPList.append(p)
+                        else:
+                            waitPList = deal_old(waitPList)
+                            waitPList.append(p)
```

Comment:
can we make this a function?

Reply:
@yihong0618 I think many places need to be refactored. A lot of code needs to be moved from loader to translator.
The same is true for https://github.com/yihong0618/bilingual_book_maker/pull/136,
but I am not familiar with python, so I don’t know how to organize it better..,